pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -809,18 +812,44 @@ async def run(self):
             self.runtime.logger.info('Building only where source has changed.')
             await self._plan_builds()
 
-        await self._rebase_and_build_rpms()
+        await self._run_with_distgit_rebase_lock(self._rebase_and_build_rpms())
+
         if not self.skip_plashets:
             await self._build_compose()
         else:
             self.runtime.logger.warning('Skipping plashets creation as SKIP_PLASHETS was set to True')
-        await self._rebase_images()
+
+        await self._run_with_distgit_rebase_lock(self._rebase_images())
+
         await self._build_images()
         await self._sync_images()
         await self._mirror_rpms()
         await self._sweep()
         self._report_success()
 
+    async def _run_with_distgit_rebase_lock(self, coro: coroutine):
```

Comment:
I like the idea of having a func like this, why not make it generic `run_with_lock()` ?

Reply:
Liked the idea, implemented with https://github.com/openshift-eng/aos-cd-jobs/pull/3886/commits/56a2786e405c7716e73efb53511950a89f3d36e9